### ISSUE TITLE
After the HMR link is closed, automatically retry connecting.

### ISSUE THREAD

**Issue by skywalker512** [farm-fe/farm#954]
### What problem does this feature solve?

When developing with vscode remote, sometimes the hmr WebSocket will disconnect. I hope it can automatically retry like vite.

https://github.com/vitejs/vite/blob/v5.0.12/packages/vite/src/client/client.ts#L107

### What does the proposed API look like?

N/A

**Comment by skywalker512** [comment]
It seems to have already been there, but it doesn't seem to be running. I don't know if it's a problem with my environment.

https://github.com/farm-fe/farm/blob/v0.16.4/packages/runtime-plugin-hmr/src/hmr-client.ts#L60

**Comment by ErKeLost** [comment]
I'll try your question.

**Comment by ErKeLost** [comment]
What does vscode remote mean? I don't quite understand how this remote is developed.

**Comment by skywalker512** [comment]
I'm very sorry, I came to the wrong conclusion. I tried using Vite in VSCode Remote and encountered the same issue.

This request will keep waiting. 

<img width="1191" alt="image" src="https://github.com/farm-fe/farm/assets/12960906/f9ef9284-4924-453f-92e4-025466d4a74b">

vscode did not experience a normal TCP reset. I found the relevant issue. 

https://github.com/microsoft/vscode-remote-release/issues/3561


> What does vscode remote mean

What I mean is to use the Forwarded Ports feature of vscode ssh remote to proxy the ports of the remote server to local.
